src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -101,44 +101,73 @@ public bool HasFlushedFinalBlock
         // cs.FlushFinalBlock() // which can only be called once
         // byte[] ciphertext = ms.ToArray();
         // cs.Close();
-        public void FlushFinalBlock()
+        public void FlushFinalBlock() =>
+            FlushFinalBlockAsync(useAsync: false).GetAwaiter().GetResult();
```

Comment:
Is this the dreaded sync over async? Or does your `useAsync:false` mean "the task isn't really async, so this isn't bad"?

Reply:
> Is this the dreaded sync over async?

Thankfully, it's not.

>  Or does your useAsync:false mean "the task isn't really async, so this isn't bad"?

Correct.  This is the same pattern followed elsewhere in the class to avoid logic duplication, so I just followed that for consistency.  When useAsync:false, the method will always perform synchronous operations and complete synchronously, so the `GetAwaiter().GetResult()` is there just to propagate any exception that may have occurred.